libobs-opengl/gl-shader.c

```diff
@@ -514,6 +514,12 @@ static void program_set_param_data(struct gs_program *program,
 			gl_success("glUniform4fv");
 		}
 
+	} else if (pp->param->type == GS_SHADER_PARAM_MATRIX3X3) {
```

Comment:
You didn't add a matrix2x2 parameter handler?

Reply:
We don't have a `struct matrix2` yet, I don't need it and see it as outside of the scope of this pull request.

Reply:
What is the purpose of matrix2x2 parameters in this pull request, if its not to enable passing a matrix2x2 parameter?

Reply:
This pull request helps obs-shaderfilter detecting types used so that it can provide better guidance on how to make the shader work for OBS.